mava/core_jax.py

```diff
@@ -0,0 +1,121 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Core Mava interfaces for Jax systems."""
+
+import abc
+from types import SimpleNamespace
+from typing import Any, Dict, List
+
+
+class BaseSystem(abc.ABC):
+    """Abstract system object."""
+
+    @abc.abstractmethod
+    def design(self) -> SimpleNamespace:
```

Comment:
There is probably a reason for this, but I am just wondering why this design cannot just happen in init?

Reply:
The main reason is that there will be things happening in the init that should be in every system for which the user should not have to care about. Therefore, separating out the components specification using the design function made more sense. Plus, this way we can more explicitly make it an abstract function to ensure it must be overwritten. This all might make more sense if you look at the general system class here: #425